src/apps/vpn/notificationhandler.cpp

```diff
@@ -196,13 +197,32 @@ void NotificationHandler::showNotification() {
           return;
         }
         // "VPN Disconnected"
-        notifyInternal(None,
-                       I18nStrings::instance()->t(
-                           I18nStrings::NotificationsVPNDisconnectedTitle),
-                       I18nStrings::instance()
-                           ->t(I18nStrings::NotificationsVPNDisconnectedMessage)
-                           .arg(localizedCityName),
-                       NOTIFICATION_TIME_MSEC);
+        ServerData* serverData = vpn->serverData();
+        if (serverData->multihop()) {
+          QString localizedEntryCityName =
+              vpn->controller()->currentServer().localizedEntryCityName();
+
+          QString localizedExitCityName =
```

Comment:
You already have this, and it's called `localizedCityName`. Use it instead of creating a new string.

If you dont like the name, you can rename it to `localizedExitCityName` at line 112.